packages/components/src/__next/context/component-system-provider.js

```diff
@@ -1,13 +1,20 @@
 /**
  * External dependencies
  */
-import { ContextSystemProvider } from '@wp-g2/components';
+import { ContextSystemProvider } from '@wp-g2/context';
 
+/**
+ * @param {Object} props
+ * @param {string[]} [props.__unstableNextInclude]
+ * @param {import('react').ReactNode} props.children
+ * @param {any} props.value
+ */
 export function ComponentSystemProvider( {
 	__unstableNextInclude = [],
 	children,
 	value = {},
 } ) {
+	// @ts-ignore
 	if ( process.env.COMPONENT_SYSTEM_PHASE === 1 ) {
```

Comment:
I know this isn't introduced here, but adding arbitrary `process.env[ whatever ]` checks to code that will ship to the browser makes it more difficult for third parties to use the package. It's fine in Gutenberg because the build system is set up for it. The problem is that it forces others to have the same build setup to use the packages containing this code. They must have this or similar: 

https://github.com/WordPress/gutenberg/blob/ec839611b9958568ac9d3d1a04fd27e09696cc25/webpack.config.js#L118-L136

It should be clearly mentioned in the package README.

It's the same case as this PR: https://github.com/WordPress/gutenberg/pull/16679

Reply:
cc @itsjonq @gziolo